src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -393,54 +397,22 @@ public String calculateSeqColLevelOneMapDigest(Map<String, String> seqColLevelOn
         return levelZeroDigest;
     }
 
-    private boolean onlyDigits(String str) {
-        String regex = "[0-9]+";
-        Pattern p = Pattern.compile(regex);
-        if (str == null) {
-            return false;
-        }
-        Matcher m = p.matcher(str);
-        return m.matches();
-    }
-
-    /**
-     * Check whether the given list contains only digits (in a form of strings)*/
-    private boolean onlyDigitsStringList(List<String> list) {
-        return list.isEmpty() || list.stream()
-                                     .allMatch(this::onlyDigits);
-    }
-
     /**
      * Return a normalized string representation of the given seqColL2Attribute
-     * Note: This is the same method as the toString of the JSONExtData class*/
-    private String convertSeqColLevelTwoAttributeValuesToString(List<String> seqColL2Attribute) {
-        StringBuilder objectStr = new StringBuilder();
-        objectStr.append("[");
-        if (onlyDigitsStringList(seqColL2Attribute)) { // Lengths array, No quotes "...". Eg: [1111, 222, 333]
-            for (int i=0; i<seqColL2Attribute.size()-1; i++) {
-                objectStr.append(seqColL2Attribute.get(i));
-                objectStr.append(",");
-            }
-            objectStr.append(seqColL2Attribute.get(seqColL2Attribute.size()-1));
-            objectStr.append("]");
-        } else { // Not a lengths array. Include quotes. Eg: ["aaa", "bbb", "ccc"].
-            for (int i=0; i<seqColL2Attribute.size()-1; i++) {
-                objectStr.append("\"");
-                objectStr.append(seqColL2Attribute.get(i));
-                objectStr.append("\"");
-                objectStr.append(",");
-            }
-            objectStr.append("\"");
-            objectStr.append(seqColL2Attribute.get(seqColL2Attribute.size()-1));
-            objectStr.append("\"");
-            objectStr.append("]");
+     * //TODO: we can find a better way to identify the given type in a more generic way*/
+    private String convertSeqColLevelTwoAttributeValuesToString(List<?> seqColL2Attribute, SeqColExtendedDataEntity.AttributeType type) {
+        switch (type) {
+            case lengths: // List<Integer> type
+                return JSONIntegerListExtData.toString((List<Integer>) seqColL2Attribute);
         }
-        return objectStr.toString();
+        // List<String> types
```

Comment:
Can make this the `default` case in the switch statement (it's the same thing just seems nicer to me)

Reply:
Done.